src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -1115,5 +1117,83 @@ void M()
 
             Assert.False(comp.NullableSemanticAnalysisEnabled);
         }
+
+        [Fact]
+        public void SymbolInfo_Invocation_InferredArguments()
+        {
+            var source = @"
+class C
+{
+    T Identity<T>(T t) => t;
+
+    void M(string? s)
+    {
+        _ = Identity(s);
+        if (s is null) return;
+        _ = Identity(s);
+    }
+}";
+
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+
+            var syntaxTree = comp.SyntaxTrees[0];
+            var root = syntaxTree.GetRoot();
+            var model = comp.GetSemanticModel(syntaxTree);
+
+            var invocations = root.DescendantNodes().OfType<InvocationExpressionSyntax>().ToArray();
+
+            var symbolInfo = model.GetSymbolInfo(invocations[0]);
+            verifySymbolInfo((IMethodSymbol)symbolInfo.Symbol, PublicNullableAnnotation.Annotated);
+            symbolInfo = model.GetSymbolInfo(invocations[1]);
+            verifySymbolInfo((IMethodSymbol)symbolInfo.Symbol, PublicNullableAnnotation.NotAnnotated);
+
+            static void verifySymbolInfo(IMethodSymbol methodSymbol, PublicNullableAnnotation expectedAnnotation)
```

Comment:
>static void verifySymbolInfo [](start = 12, length = 28)

In some future PR, would it be worth updating `VerifyTypes()` to allow specifying the expectations in code?
```
_ = Identity(s) /*S:string! Identity<string!>(string! t)*/
```